### PR TITLE
Use the String#delete_prefix method introduced in Ruby 2.5

### PR DESCRIPTION
**Pull request by ento**
Follow-up to #26: updates target Ruby version to 2.5 to be able to use `String#delete_prefix`.

Also updates Bundler to 2.x, motivated by the fact that the version of Bundler included in the revision of nixpkgs I picked (the latest nixpkgs-darwin-20.03pre as of this writing) was 2.x.

### CODE REVIEW

File: spec/lib/deploy_complexity/git_spec.rb [hunk 1]
```diff
@@ -7,8 +7,11 @@
   describe ".safe_name" do
     subject { DeployComplexity::Git.method(:safe_name) }
     its(['origin/master']) { is_expected.to eq 'master' }
+    its(["origin/master\n"]) { is_expected.to eq 'master' }
+    its(["origin/master\n\n"]) { is_expected.to eq "master\n" }
```
Comment (ento) [inline spec/lib/deploy_complexity/git_spec.rb:11]: This can be surprising, though I'm not sure if the use of `#chomp` over `#strip` was intentional.

Comment (BrianHicks) [reply spec/lib/deploy_complexity/git_spec.rb:11]: it wasn't! It should probably be `#strip`.

**Comment by ento** [comment]
👍 Switched to `strip`, will merge when CI is green. I think I'll make another PR to bump the version..

**Comment by ento** [comment]
> I think I'll make another PR to bump the version..

Circling back to this cliffhanger: https://github.com/NoRedInk/deploy-complexity/pull/28 is the PR and I asked Jasper/Charles for a review.

**Pull request merged**